sphinxgallery/gen_rst.py

```diff
@@ -444,7 +458,12 @@ def generate_file_rst(fname, target_dir, src_dir):
     ref_fname = example_file.replace(os.path.sep, '_')
     example_rst = """\n\n.. _sphx_glr_{0}:\n\n""".format(ref_fname)
 
-    if not fname.startswith('plot'):
+    exec_script = fname.startswith('plot')
+    use_mayavi = gallery_conf.get('use_mayavi', False)
+    if not use_mayavi and any('mayavi' in bl[1] for bl in script_blocks):
```

Comment:
> Otherwise the demo gallery does not build fine if mayavi is not present.

That's the part I don't really understand, does that affect other examples, e.g. because the mayavi import segfaults and the sphinx generation is stopped forcefully? In other words, how is mayavi not being installed different from a SyntaxError in the example?